src/main/java/racingcar/domain/name/Names.java

```diff
@@ -0,0 +1,30 @@
+package racingcar.domain.name;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import racingcar.exception.name.NameException;
+import racingcar.exception.name.NameExceptionMessage;
+
+public class Names {
+	private final List<Name> names;
+
+	public Names(List<Name> names) {
+		validateNames(names);
+		this.names = names;
+	}
+
+	public List<Name> getNames() {
+		return names;
+	}
```

Comment:
기본적으로 전제가 names는 불변이라는 가정을 하고 만든 클래스인 것 같습니다.
하지만 코드처럼 단순히 names를 반환하게 되면 List의 주소가 넘어갑니다. 따라서, 외부에서 names.add(null)을 하게 되면 그대로 입력되게 됩니다.
즉, 검증되지 않은 데이터가 names에 포함되게 됩니다.
아래와 같이 넘겨주게되면 Names의 불변성을 보장해줄 수 있게 됩니다.

```suggestion
	public List<Name> getNames() {
		return new ArrayList<>(names);
	}
```

Reply:
사실 위의 방법으로 한다고 해도 불변성이 보장되지는 않을 것 같습니다. Collectors.toUnmodifiableList()를 사용하는게 더 맞지 않을까 싶긴 하네요. 그래도 피드백의 의도는 좋은것 같네요!

Reply:
저는 반환하는 `List의 원소`까지 불변으로 보장해줄 필요가 있나 싶습니다. 만약 외부에서 해당 값들을 조작하고 싶다면 다시 값을 옮겨담아서 조작해야하니까요. 제가 의도한 `불변`이라는 것은 클래스내의 원소를 외부에서 임의로 조작하지 못한다는 의미였습니다. 위와 같이 반환하면 내용을 전달해줄 뿐이지 클라이언트가 해당 `List`를 어떻게 조작하든 상관없으니까요!

그런 의미에서 `ArrayList`로 반환하는 것은 오히려 좋은 방법이라고 할 수 있죵

Reply:
다음 주차에 반영해보겠습니다 감사합니다!